compiler/rustc_hir/src/definitions.rs

```diff
@@ -281,6 +294,10 @@ pub enum DefPathData {
     AnonConst,
     /// An `impl Trait` type node.
     ImplTrait,
+    /// An `impl Trait` type node inside a `trait` or inside an `impl` of a `trait`.
+    /// On a `trait` the value is `(trait_fn_def_id, None)`.
+    /// On an `impl` the value is `(impl_fn_def_id, Some(trait_rpit_def_id))`.
+    ImplTraitInTrait(DefId, Option<DefId>),
```

Comment:
This may cause issues with incremental compilation.  `DefPathData` is hashed using `Hash` and not `StableHasher`.  This means that we may change the `DefPathHash` when `DefIndices` are re-numbered, which is what `DefPathHash` is supposed to avoid.